### PR TITLE
feat(flexibleText): #INFR-7980  replace TooltipService with ThyTooltipService in thy-flexible-text

### PR DESCRIPTION
**Pull request by mengshuicmq**

### CODE REVIEW

File: src/flexible-text/flexible-text.component.ts [hunk 1]
```diff
@@ -110,15 +116,30 @@ export class ThyFlexibleTextComponent implements OnInit, AfterContentInit, OnDes
 
     ngOnInit() {
         this.updateContainerClass();
-        this.tooltipService.attach(this.elementRef, this.viewContainerRef, this.trigger);
+        this.attachTooltip();
+    }
+
+    attachTooltip() {
+        this.thyTooltipDirective = new ThyTooltipDirective(
```
Comment (minlovehua) [inline src/flexible-text/flexible-text.component.ts:123]: this.tooltipRef = this.thyTooltipService.create(this.elementRef, {
         viewContainerRef: this.viewContainerRef,
         placement: this.placement,
         offset: this.offset,
         hasBackdrop: this.trigger === 'click',
         // ...
});

File: src/flexible-text/flexible-text.component.ts [hunk 2]
```diff
@@ -110,15 +116,30 @@ export class ThyFlexibleTextComponent implements OnInit, AfterContentInit, OnDes
 
     ngOnInit() {
         this.updateContainerClass();
-        this.tooltipService.attach(this.elementRef, this.viewContainerRef, this.trigger);
+        this.attachTooltip();
```
Comment (minlovehua) [inline src/flexible-text/flexible-text.component.ts:119]: 重命名为 createTooltipRef ，或者不需要新增 attachTooltip 方法

File: src/flexible-text/flexible-text.component.ts [hunk 3]
```diff
@@ -110,15 +116,30 @@ export class ThyFlexibleTextComponent implements OnInit, AfterContentInit, OnDes
...
+            this.elementRef,
+            this.ngZone,
+            this.platform,
+            this.focusMonitor,
+            this.viewContainerRef,
+            this.thyTooltipService
+        );
         if (this.placement) {
-            this.tooltipService.thyTooltipDirective.placement = this.placement;
+            this.thyTooltipDirective.placement = this.placement;
```
Comment (minlovehua) [inline src/flexible-text/flexible-text.component.ts:132]: 下面这些都移除掉，不需要了：

image

File: src/flexible-text/flexible-text.component.ts [hunk 4]
```diff
@@ -148,7 +169,7 @@ export class ThyFlexibleTextComponent implements OnInit, AfterContentInit, OnDes
 
     ngOnDestroy() {
         this.destroy$.next();
-        this.tooltipService.detach();
+        this.thyTooltipDirective.ngOnDestroy();
```
Comment (minlovehua) [inline src/flexible-text/flexible-text.component.ts:172]: this.tooltipRef?.dispose();

File: src/flexible-text/flexible-text.component.ts [hunk 5]
```diff
@@ -158,7 +179,7 @@ export class ThyFlexibleTextComponent implements OnInit, AfterContentInit, OnDes
         } else {
             this.isOverflow = false;
         }
-        this.tooltipService.thyTooltipDirective.thyTooltipDisabled = !this.isOverflow;
+        this.thyTooltipDirective.thyTooltipDisabled = !this.isOverflow;
```
Comment (minlovehua) [inline src/flexible-text/flexible-text.component.ts:182]: 不需要依赖 thyTooltipDirective 指令了，移除

File: src/flexible-text/flexible-text.component.ts [hunk 6]
```diff
@@ -158,7 +179,7 @@ export class ThyFlexibleTextComponent implements OnInit, AfterContentInit, OnDes
         } else {
             this.isOverflow = false;
```
Comment (minlovehua) [inline src/flexible-text/flexible-text.component.ts:180]: 删除 isOverflow 变量。
使用 this.tooltipRef.show(this.content); 替代  this.isOverflow=true 
使用 this.tooltipRef.hide(); 替代 this.isOverflow = false;

**Review by minlovehua — CHANGES_REQUESTED** [review]

File: src/flexible-text/examples/trigger/trigger.component.html
Comment (mengshuicmq) [inline src/flexible-text/examples/trigger/trigger.component.html:1]: 示例已跟设计师确认
image

File: src/flexible-text/examples/trigger/trigger.component.html [hunk 7]
```diff
@@ -0,0 +1,26 @@
+<div class="d-flex">
```
Comment (walkerkay) [inline src/flexible-text/examples/trigger/trigger.component.html:1]: 这个Demo叫 Trigger 不是特别好，可以是 WithTooltip

File: src/flexible-text/examples/trigger/trigger.component.html [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+  <span>触发方式分为：</span>
```
Comment (walkerkay) [inline src/flexible-text/examples/trigger/trigger.component.html:2]: 我看了看Demo 感觉这个Demo布局不太好，你在考虑考虑优化一下

**Review by walkerkay — COMMENTED** [review]
实现没问题，Demo感觉可以再优化优化

File: src/flexible-text/examples/trigger/trigger.component.html
Comment (walkerkay) [reply src/flexible-text/examples/trigger/trigger.component.html:1]: > 示例已跟设计师确认 image

文案优化为：Feedback 的触发方式有三种
既然已经加了 Label 了， 就别加「、」了，加个空格隔开就可以了
